packages/base/skill-card.gts

```diff
@@ -0,0 +1,25 @@
+import BooleanField from './boolean';
+import {
+  CardDef,
+  Component,
+  field,
+  FieldDef,
+  contains,
+  linksTo,
+} from './card-api';
+import TextAreaField from './text-area';
+
+export class SkillCard extends CardDef {
+  static displayName = 'Skill';
+  @field instructions = contains(TextAreaField);
+  static embedded = class Embedded extends Component<typeof this> {
+    <template>
+      <@fields.title />
+    </template>
+  };
+}
+
+export class SkillField extends FieldDef {
+  @field isActive = contains(BooleanField);
+  @field card = linksTo(SkillCard);
+}
```

Comment:
This should use the same mechanism for attaching cards, that is that the skill gets serialised and attached in the messages. If the serialisation is the same as previously, it points to the same event, otherwise it updates a new version.

I wonder if this is down to where gets updates? We can file a bug for this, and see if it happens similarly with attached cards.

Reply:
when the skill is turned off, we remove it from the attached skills that's sent with the message. that's the slight difference from attached cards, which are cumulative. however, even though it's removed, AI still can reference it if it wants to, since it's part of the past room events, which we do not change, so turning a skill off is not 100% guaranteed. this was the main reason we only have a 'turn off' switch instead of a remove button for attached skills, since you can't really remove the knowledge fully once it's been attached, unless you start a new session.